src/proposer/proposer.cpp

```diff
@@ -178,13 +228,30 @@ class ProposerImpl : public Proposer {
     m_waiter.Wake();
   }
 
+  bool GenerateBlock(CWallet *wallet,
+                     const CBlockIndex &tip,
+                     const staking::CoinSet &coins,
+                     const boost::optional<CScript> &coinbase_script,
+                     std::shared_ptr<const CBlock> &block_out) override {
```

Comment:
This function invokes `BuildBlock` which returns `std::shared_ptr<const CBlock>`. Why don't keep the same semantic and return `std::shared_ptr<const CBlock>` instead of `bool + out parameter`?

Reply:
yeah, gonna commit that in a sec, already changed it.